[EDIT] Added an ability to have custom WSSE usernametoken attribute. Added unit testing for it.

ohpyupi opened the pull request:
I needed to put custom attribute to usernamToken element in WSSE security header like below.
<wsse:UsernameToken kor:clientId="myClientId" xmlns:kor="https://www.korea.com/auth">
...
</wsse:UsernameToken>

It will be achieved by passing additional keys and properties to options for soap.WSSecurity constructor.

For example,
let options = {
...
customWsuName: 'kor',
customWsuId: 'clientId',
customWsuValue: 'myClientId',
customWsuPath: 'https://www.korea.com/auth',
};

Unless specified, default values will be used.

herom commented:
Thanks a lot for your PR @ohpyupi 👍 

Please see our Guidelines on  [Submitting a Pull Request](https://github.com/vpulim/node-soap/blob/master/CONTRIBUTING.md#submitting-a-pull-request) in our [CONTRIBUTING.md](https://github.com/vpulim/node-soap/blob/master/CONTRIBUTING.md) file and add at least(!) 1 test which will help us guarantee that your PR is not overwritten accidentially with another PR in the future.

herom commented on lib/security/WSSecurity.js:
```diff
@@ -1,90 +1,97 @@
-};
-
-module.exports = WSSecurity;
+"use strict";
```
why was the whole file updated? could you please revert and only change the code you actually re-wrote/added?

ohpyupi replied:
I think that's because I copied and pasted the file entirely.
Yeap! I will go through your guideline and follow your feedback!

herom replied:
thanks 👍

ohpyupi commented:
Added a unit testing!
Also, I am using vim editor and seems like indentation causes some problem to jshint. So when I was testing the program in my local machine, I commented out indent option in .jshint file.

herom commented:
@ohpyupi sorry for the delay - somehow I did manage to completely oversee the notifcation for this PR changes 😞 

I'll just step over the commited changes right now 👍

herom commented on lib/security/WSSecurity.js:
```diff
@@ -75,16 +70,28 @@ WSSecurity.prototype.toXML = function() {
 
 module.exports = WSSecurity;
+
+function getDate(d) {
```
would you mind to move this function to the top of the file alongside all the other declarations so we don't have things implemented/positioned _after_ the call to `module.exports`?

herom commented on test/security/WSSecurity.js:
```diff
@@ -69,4 +70,37 @@ describe('WSSecurity', function() {
+		});
+	});
 });
+
```
would you mind removing this unnecessary blank lines?

herom commented on test/custom-usernametoken-attr-test.js:
```diff
@@ -0,0 +1,5 @@
+'use strict';
+
+describe('Custom UsernameToken Attribute', function () {
+
```
are you planning on implementing this test? this would be awesome 👍

herom commented on test/security/WSSecurity.js:
```diff
@@ -69,4 +70,37 @@ describe('WSSecurity', function() {
   });
+	var username = 'myUser';
+	var password = 'myPass';
+	it('should accept additional parameters for custom UsernameToken attribute.', function() {
```
please move the `var`-declaration *inside* the test's `function` scope

herom commented on test/security/WSSecurity.js:
```diff
@@ -1,7 +1,8 @@
 var fs = require('fs'),
-  join = require('path').join;
+  join = require('path').join,
+	assert = require('assert');
```
please remove the unused `assert` lib `require`

ohpyupi replied:
Oh sorry for the late response. Yeah I will work on your feedback!

ohpyupi commented:
@herom I addressed the comments left. But the Travis build fails. I ran `npm run cover` on master branch but the got the same error. Any idea how to resolve this? Also, are you guys not using `jshint` anymore? `npm run pretest` gives errors too. Curios if its my local machine issue.
